src/NHibernate/Hql/Ast/HqlTreeNode.cs

```diff
@@ -257,6 +257,30 @@ internal HqlIdent(IASTFactory factory, System.Type type)
 					throw new NotSupportedException(string.Format("Don't currently support idents of type {0}", type.Name));
 			}
 		}
+
+		internal static bool SupportsType(System.Type type)
+		{
+			type = type.UnwrapIfNullable();
+			switch (System.Type.GetTypeCode(type))
+			{
+				case TypeCode.Boolean:
+				case TypeCode.Int16:
+				case TypeCode.Int32:
+				case TypeCode.Int64:
+				case TypeCode.Decimal:
+				case TypeCode.Single:
+				case TypeCode.DateTime:
+				case TypeCode.String:
+				case TypeCode.Double:
+					return true;
+				default:
+					return new[]
```

Comment:
Probably better to replace with just if-elif-else.

Reply:
I've simplified it by using the `||` operator.